chia/full_node/full_node.py

```diff
@@ -699,7 +699,10 @@ async def short_sync_backtrack(
                         f"Failed to fetch block {curr_height} from {peer.get_peer_logging()}, wrong type {type(curr)}"
                     )
                 blocks.append(curr.block)
-                if self.blockchain.contains_block(curr.block.prev_header_hash) or curr_height == 0:
+                if (
+                    self.blockchain.contains_block(curr.block.prev_header_hash, curr.block.height - 1)
+                    or curr_height == 0
+                ):
```

Comment:
I agree that this call site is not concerned with whether the block is in the cache or not